### PR TITLE
resource/wafv2_web_acl and resource/wafv2_rule_group: rewrite typeset checks

### PR DESCRIPTION
**Pull request by anGie44**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource/wafv2_web_acl and resource/wafv2_rule_group: rewrite typeset checks
```

Output from acceptance testing:

```
--- PASS: TestAccAwsWafv2RuleGroup_Disappears (172.09s)
--- PASS: TestAccAwsWafv2RuleGroup_Minimal (234.66s)
--- PASS: TestAccAwsWafv2RuleGroup_RegexPatternSetReferenceStatement (264.38s)
--- PASS: TestAccAwsWafv2RuleGroup_IpSetReferenceStatement (277.72s)
--- PASS: TestAccAwsWafv2RuleGroup_XssMatchStatement (58.28s)
--- PASS: TestAccAwsWafv2RuleGroup_ChangeCapacityForceNew (461.62s)
--- PASS: TestAccAwsWafv2RuleGroup_ChangeMetricNameForceNew (464.85s)
--- PASS: TestAccAwsWafv2RuleGroup_ChangeNameForceNew (466.92s)
--- PASS: TestAccAwsWafv2RuleGroup_GeoMatchStatement (538.54s)
--- PASS: TestAccAwsWafv2RuleGroup_SizeConstraintStatement (539.88s)
--- PASS: TestAccAwsWafv2RuleGroup_SqliMatchStatement (542.53s)
--- PASS: TestAccAwsWafv2RuleGroup_ByteMatchStatement (542.88s)
--- PASS: TestAccAwsWafv2RuleGroup_Basic (566.92s)
--- PASS: TestAccAwsWafv2RuleGroup_Tags (671.91s)
--- PASS: TestAccAwsWafv2RuleGroup_RuleAction (800.63s)
--- PASS: TestAccAwsWafv2RuleGroup_LogicalRuleStatements (823.80s)
--- PASS: TestAccAwsWafv2RuleGroup_ByteMatchStatement_FieldToMatch (1850.58s)
```
```
--- PASS: TestAccAwsWafv2WebACL_Disappears (2349.71s)
--- PASS: TestAccAwsWafv2WebACL_Minimal (3189.81s)
--- PASS: TestAccAwsWafv2WebACL_MaxNestedRateBasedStatements (3537.71s)
--- PASS: TestAccAwsWafv2WebACL_MaxNestedOperatorStatements (3491.60s)
--- PASS: TestAccAwsWafv2WebACL_ChangeNameForceNew (5178.04s)
--- PASS: TestAccAwsWafv2WebACL_ManagedRuleGroupStatement (5550.41s)
--- PASS: TestAccAwsWafv2WebACL_RateBasedStatement (5568.75s)
--- PASS: TestAccAwsWafv2WebACL_Basic (5729.94s)
--- PASS: TestAccAwsWafv2WebACL_RuleGroupReferenceStatement (5673.26s)
--- PASS: TestAccAwsWafv2WebACL_Tags (5803.29s)
```
```
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_webACLDisappears (2838.28s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_basic (3273.20s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_disappears (3360.32s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_changeLogDestinationConfigsForceNew (4724.55s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_changeResourceARNForceNew (5311.34s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_update (5743.55s)
```
```
--- PASS: TestAccAwsWafv2WebACLAssociation_Basic (3285.04s)
--- PASS: TestAccAwsWafv2WebACLAssociation_Disappears (4970.96s)
```

### CODE REVIEW

**Comment by bflad** [comment]
@anGie44 can you rebase this please and I'll take a look? Thanks! 😄

File: aws/resource_aws_wafv2_rule_group_test.go [hunk 1]
```diff
@@ -56,29 +56,34 @@ func TestAccAwsWafv2RuleGroup_Basic(t *testing.T) {
...
+						"action.0.count.#": "0",
+						"statement.#":      "1",
+						"statement.0.size_constraint_statement.#":                                 "1",
+						"statement.0.size_constraint_statement.0.comparison_operator":             "LT",
+						"statement.0.size_constraint_statement.0.field_to_match.#":                "1",
+						"statement.0.size_constraint_statement.0.field_to_match.0.query_string.#": "1",
+						"statement.0.size_constraint_statement.0.size":                            "50",
+						"statement.0.size_constraint_statement.0.text_transformation.#":           "2",
+						// TODO: super nested TypeSet check
+						"statement.0.size_constraint_statement.0.text_transformation.2212084700.priority": "2",
```
Comment (appilon) [inline aws/resource_aws_wafv2_rule_group_test.go:74]: So not a perfect setup but when dealing with a deeply nested set, validate that set independently of it's outer set

```
tfawsresource.TestCheckTypeSetElemNestedAttrs(resourceName, "rule.*.statement.0.size_constraint_statement.0.text_transformation.*", map[string]string{
    "priority": "2"
    // ...
},
```

Comment (anGie44) [reply aws/resource_aws_wafv2_rule_group_test.go:74]: ahh yesss!

Comment (appilon) [reply aws/resource_aws_wafv2_rule_group_test.go:74]: The risk here is false positive for an inner set on "the wrong" outer set.... I have a concrete example in the AWS codebase somewhere but I wouldn't worry about it for sets with high specificity

**Pull request merged**